src/Service/HealthCheck/HealthCheckHelper.cs

```diff
@@ -57,6 +61,38 @@ public ComprehensiveHealthCheckReport GetHealthCheckResponse(HttpContext context
             return ComprehensiveHealthCheckReport;
         }
 
+        // Updates the incoming role header with the appropriate value from the request headers.
+        public void UpdateIncomingRoleHeader(HttpContext httpContext)
```

Comment:
nit: consider renaming this function to `StoreIncomingRoleHeader` since we are storing the role header from the httpContext to a private member. `Update` sounds like we are updating the role header in the context itself since external functions shouldnt be knowing private member names.